packages/react-utils/src/createRequiredContext.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react'
+
+/**
+ * Create common context-related values from React with a required context.
+ *
+ * @example
+ * ```typescript
+ * const [useFoo, FooConsumer, FooProvider] = createRequiredContext<FooContextValue, 'Foo'>('Foo')
+ * ```
+ */
+
+export default function createRequiredContext<TContextProps, TName extends string>(
+  contextName: TName,
+) {
+  const context = React.createContext<null | TContextProps>(null)
+
+  if (process.env.NODE_ENV !== 'production') {
+    context.displayName = `${contextName}Context`
+  }
+
+  const providerName = `${contextName}Provider` as const
+  const hookName = `use${contextName}` as const
```

Comment:
nitpick: If these variables are only used for the error they might as well be defined within the if statements scope (`if (contextValue === null) {`).

suggestion: How would it look/feel if these were defined in a one-liner as below with only a single `as const`?
```ts
const [hookName, providerName] = [`use${contextName}`, `${contextName}Provider`] as const
```